### ISSUE TITLE
Think about authentication and identity management

### ISSUE THREAD

**Issue by sebgoa** [vmware-archive/kubeless-ui#2]
We need to think about authentication for this.

The k8s dashboard might give us some ideas.

The coreOS dex project might be useful as well.

**Comment by serverlessnomad** [comment]
I would also advocate for adding authentication.  I deployed the UI after standing up a cluster via Stackpoint only to find it open to the world and having to take it down.  I'll be looking for an alternative way to leverage the UI because I like the utility of it, bit having some auth in front would be good.  Is the expectation that the UI sits only on an internal network and would never be accessed over the Internet?

**Comment by sebgoa** [comment]
The problem is that to create a function through the UI, one needs to pass its k8s creds somehow otherwise the service account running the UI needs "cluster wide" access.

This is the same issue as the k8s dashboard

**Comment by sebgoa** [comment]
Godo wiki doc here: https://github.com/kubernetes/dashboard/wiki/Access-control

**Comment by thoughtentity** [comment]
Definitely, would like to see `Open ID Connect` support. It is one of the [natively supported](https://kubernetes.io/docs/admin/authentication/#openid-connect-tokens) authentication strategies in Kubernetes that supports external identity provider logins (`Auth0`, `Google` and `GitHub` among others). Kubernetes's `Dashboard` is tacking support in kubernetes/dashboard#2533, however, they're waiting for some APIs to land in Core in order to query the configured authentication strategies in Kubernetes.

As far as being secure out of the box. I'd recommend changeing the service type to `ClusterIP` and just using `kubectl proxy` to open a secure channel into the cluster.

**Comment by sebgoa** [comment]
@thoughtentity yes good points.

We are actually re-writing a kubeless UI in kubeapps: https://github.com/kubeapps/kubeapps . the dashboard is indeed opened via a proxy connection, plus we are starting to align with the k8s dashboard auth strategies:

https://github.com/kubeapps/kubeapps/issues/199

PR welcome on any projects to move in the right direction :)
